pkg/kademlia/kademlia.go

```diff
@@ -336,7 +336,13 @@ func (k *Kad) connect(ctx context.Context, peer swarm.Address, ma ma.Multiaddr,
 	i, err := k.p2p.Connect(ctx, ma)
 	if err != nil {
 		if errors.Is(err, p2p.ErrAlreadyConnected) {
-			return nil
+			if i == nil || swarm.ZeroAddress.Equal(i.Overlay) || i.Overlay.Equal(peer) {
+				return nil
```

Comment:
if the peer is already connected, how could it be that `i` is `nil` or `ZeroAddress`?

Reply:
Originally the `i` was not being returned when error (i.e. `p2p.ErrAlreadyConnected`) was response from `Connect` call. Since one of the change here now returns `bzz.Address` on that error, it should be fine to remove additional checks. I had them while testing possible solutions; will remove it.